### PR TITLE
csrf: register type for gob serialization

### PR DESCRIPTION
**Pull request by wuhan005**
### Describe the pull request

Fix https://github.com/flamego/flamego/issues/152

Link to the issue:  https://github.com/flamego/flamego/issues/152

### Checklist

- [x] I agree to follow the [Code of Conduct](https://go.dev/conduct) by submitting this pull request.
- [x] I have read and acknowledge the [Contributing guide](https://github.com/flamego/flamego/blob/main/.github/contributing.md).
- [x] I have added test cases to cover the new code.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -44,7 +44,42 @@ jobs:
...
+          - 6379:6379
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run tests with coverage
+        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run ^\QTestGobSerialization\E$
```
Comment (unknwon) [inline .github/workflows/go.yml:79]: I don't think this regep is working? 

```
testing: warning: no tests to run
```

https://github.com/flamego/csrf/actions/runs/3533397520/jobs/5928964766#step:5:25

**Review by unknwon — CHANGES_REQUESTED** [review]
Thanks for the fix!

File: .github/workflows/go.yml (hunk 1 above)
Comment (wuhan005) [reply .github/workflows/go.yml:79]: Yes, I've also noticed something wrong with the regex.
This PR is still a work in progress. 😥

Comment (unknwon) [inline .github/workflows/go.yml:79]:
```suggestion
        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run ^TestGobSerialization$
```

Comment (unknwon) [reply .github/workflows/go.yml]: I fixed the test run, but I don't think it makes to _not_ run the full test suite unless it becomes painfully long.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -44,7 +44,42 @@ jobs:
...
+          - 6379:6379
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run tests with coverage
+        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run ^TestGobSerialization$
```
Comment (unknwon) [inline .github/workflows/go.yml:79]:
```suggestion
        run: go test -v -race -coverprofile=coverage -covermode=atomic ./...
```

**Review by unknwon — APPROVED** [review]

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -36,6 +36,38 @@ jobs:
...
     runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run tests with coverage
+        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run !TestGobSerialization
```
Comment (unknwon) [inline .github/workflows/go.yml:47]:
```suggestion
        run: go test -v -race -coverprofile=coverage -covermode=atomic ./...
```

**Comment by unknwon** [comment]
https://github.com/flamego/csrf/releases/tag/v1.1.1 has been created for this merge.

**Pull request merged**